src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala

```diff
@@ -1018,7 +1018,9 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
     aSym.info.members.exists(s => localShouldDocument(s) && (!s.isConstructor || s.owner == aSym))
 
   def localShouldDocument(aSym: Symbol): Boolean =
-    !aSym.isPrivate && (aSym.isProtected || aSym.privateWithin == NoSymbol) && !aSym.isSynthetic
+    ((aSym.isPrivate && !aSym.isTopLevel) ||
+     (!aSym.isPrivate && (aSym.isProtected || aSym.privateWithin == NoSymbol))) &&
```

Comment:
I don't understand what the `privateWithin` check is for, can we remove it? As far as I know, `privateWithin` can be set in `private[x]` and `protected[x]`. The line excludes any private, and includes any protected, so the `privateWithin` seems useless, no? Maybe just try if all tests pass if you remove it.

Reply:
Removing `aSym.privateWithin == NoSymbol` breaks the following tests.

```
[info] Elapsed time: 0.045 sec 
[info] ! HtmlFactory.scala/bug#8144: Members' permalink - inner package: Falsified after 0 passed tests.
[info] Elapsed time: 0.000 sec 
[info] ! HtmlFactory.scala/bug#8144: Members' permalink - companion object: Falsified after 0 passed tests.
[info] Elapsed time: 0.001 sec 
[info] ! HtmlFactory.scala/bug#8144: Members' permalink - class: Falsified after 0 passed tests.
[info] Elapsed time: 0.000 sec 
[info] ! HtmlFactory.scala/bug#9599 Multiple @todo formatted with comma on separate line: Falsified after 0 passed tests.
[info] Elapsed time: 0.063 sec 
[info] ! HtmlFactory.scala/bug#10999 Private and Protected method should also be documented: Falsified after 0 passed tests.
```

Reply:
Sorry, I was not very clear.. I pushed a change that simplifies `localShouldDocument`, hopefully that passes the tests.

Reply:
OK, here is my misunderstanding:
```
class C {
  private def a
  private[C] def b
  protected def c
  protected[C] def d
}
```

We get

| | isPrivate | isProtected | privateWithin |
| --- | --- | --- | --- |
| a | true | false | NoSymbol |
| b | false | false | C |
| c | false | true | NoSymbol |
| d | false | true | C |

Note that for `b`, `isPrivate` is false, but for `d`, `isProtected` is true.